filepath_unix.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2015 Matthew Holt and The Caddy Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build !windows
```

Comment:
```suggestion
```

Not necessary, the file will be ignored on Windows because of `_unix` in the name: https://pkg.go.dev/go/build#hdr-Build_Constraints

Reply:
But is `unix` the exact inverse of `windows` in actuality? I don't think it is. But I want specifically that, to match `filepath_unix.go`.

Reply:
Your file will be ignored on non-Unix platforms anyway, because of it's name. What you likely want is:

* `filepath.go` delegating to `filepath.Abs()` and containing:
	```
	//go:build !unix
	```
* `filepath_unix.go` with the Unix-improved implementation.

Reply:
There's only unix and non-unix (windows), there's no valid non-unix base implementation.

My understanding of the underscore filenames is that it's a default build constraint, but `//go:build` takes precedence if it appears.

Reply:
> I think it would be confusing/misleading to have the Windows implementation in filepath.go

I'm not sure why that would be. :thinking: I think the build tag isn't needed here, and the filename is enough. We can tweak later if needed. Having the build tag just creates more logical complexity I think. ("Well, the file is intended for unix hosts, but the build tag is for non-Windows hosts, so ...")

Reply:
@francislavoie name if `fastabs.go` and `fastabs_unix.go` instead as I've done if you fear the confusion: https://github.com/dunglas/frankenphp/pull/1154

Reply:
Ended up renaming the file to just `filepath.go`, but the `//go:build !windows` needs to stay there.